app/src/main/java/edu/mills/cs115/fruitthief/MainActivity.kt

```diff
@@ -0,0 +1,118 @@
+package edu.mills.cs115.fruitthief
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import androidx.navigation.findNavController
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.fragment.app.add
+import androidx.fragment.app.commit
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import androidx.navigation.ui.*
+import edu.mills.cs115.fruitthief.databinding.ActivityMainBinding
+import androidx.fragment.app.add
+import androidx.fragment.app.commit
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import edu.mills.cs115.fruitthief.database.FruitTreeDatabase
+import edu.mills.cs115.fruitthief.database.PopulateFruitTable
+import edu.mills.cs115.fruitthief.database.PopulateTreeTable
+import edu.mills.cs115.fruitthief.map.MapFragment
+import kotlinx.coroutines.runBlocking
+
+class MainActivity : AppCompatActivity() { //, OnMapReadyCallback {
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var navController: NavController
+    private lateinit var drawerLayout: DrawerLayout
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        navController = findNavController(R.id.nav_host_fragment)
+        drawerLayout = binding.drawerLayout
+        val application = requireNotNull(this).application
+        val dataSource = FruitTreeDatabase.getInstance(application).fruitTreeDAO
+
+        // clear tables for testing
```

Comment:
This comment appears twice. I don't understand why you do anything for testing in `MainActivity`. Could you explain?

Reply:
We needed to clear the tables to simulate running the app for the first time. Normally the tables should keep all their info from previous times the app is used, but we didn't want the clutter from all the trees we had added before each time we needed to try a new feature.